paimon-core/src/main/java/org/apache/paimon/mergetree/MergeTreeWriter.java

```diff
@@ -283,15 +286,17 @@ private CommitIncrement drainIncrement() {
                         new ArrayList<>(compactBefore.values()),
                         new ArrayList<>(compactAfter),
                         new ArrayList<>(compactChangelog));
+        CompactDeletionFile drainDeletionFile = this.compactDeletionFile;
 
         newFiles.clear();
         deletedFiles.clear();
         newFilesChangelog.clear();
         compactBefore.clear();
         compactAfter.clear();
         compactChangelog.clear();
+        this.compactDeletionFile = null;
 
-        return new CommitIncrement(dataIncrement, compactIncrement);
+        return new CommitIncrement(dataIncrement, compactIncrement, drainDeletionFile);
```

Comment:
Can we run `compactDeletionFile.getOrCompute` here, and build a `IndexIncrement`

So that we can make `CommitIncrement` like this:
```
public class CommitIncrement {
    private final DataIncrement dataIncrement;
    private final CompactIncrement compactIncrement;
    private final IndexIncrement indexIncrement;
}
```